[docs] more emphatic direnv recommendation (#21810)
expo/expo: docs/pages/guides/environment-variables.mdx

This approach this plugin will work well with environment variables set in your shell process, but it won't automatically load **.env** files. We recommend using [direnv](https://direnv.net/) and **.envrc** to accomplish that.

## Loading environment variables from a file

> If you want to automatically load environment variables from a file in your project rather than setting them in your shell profile or manually when running `npx expo start`, you may want to use an environment variable manager.

Reviewer: ```suggestion
If you want to automatically load environment variables from a file in your project rather than setting them in your shell profile or manually when running `npx expo start`, you may want to use an environment variable manager such as direnv or dotenv.
```

Author: Eh, your bold below helps, and "such as" also helps, I'm good with it :-)

Reviewer: We can remove the phrase if you think the addition above doesn't add anything or is going to lead to confusion.

One way I think is, at times, we should spell out terms we want to communicate (it has small benefits, for example, in some cases a screen reader will pick this up or somebody skimming will go through first and last words of a paragraph, etc.) That was my intention behind it.

### direnv

> [direnv](https://direnv.net/) automatically loads and unloads environment variables in your shell depending on your current directory. When you `cd` into your project directory, it will load all of the variables as configured in **.envrc**. Due to its tight shell integration, we highly recommend direnv for the highest level of compatiblity with all Expo and EAS CLI tooling. [Learn more](https://direnv.net/#getting-started).

Reviewer: ```suggestion
[direnv](https://direnv.net/) automatically loads and unloads environment variables in your shell depending on your current directory. When you `cd` into your project directory, it will load all of the variables as configured in **.envrc**. Due to its tight shell integration, **we highly recommend direnv for the highest level of compatibility with all Expo and EAS CLI tooling**. To get started, see [direnv documentation](https://direnv.net/#getting-started).
```

### dotenv

> [dotenv](https://github.com/motdotla/dotenv) is a library that you import in your project code to load the **.env** file. Since it works at the JavaScript code level, dotenv may be affected by your Node environment or how you invoke it from your code, so be sure to check their documentation for the correct usage if you run into issues. [Learn more](https://github.com/motdotla/dotenv).

Reviewer: ```suggestion
[dotenv](https://github.com/motdotla/dotenv) is a library that you import in your project code to load the **.env** file. Since it works at the JavaScript code level, dotenv may be affected by your Node.js environment or how you invoke it from your code. Check the [documentation](https://github.com/motdotla/dotenv) for the correct usage if you run into issues.
```
